### PR TITLE
Fix image attachment links

### LINKED ISSUE #132: Image attachments in issues / issue comments do not work
as seen in https://34.105.185.241/sagemath/sage-20230114144245/issues/34113

### PR DESCRIPTION
**Pull request by kwankyu**
Hopefully fixes #132.

### CODE REVIEW

File: migrate.py [hunk 1]
```diff
@@ -1363,22 +1357,38 @@ def set_ticket_paths(self, ticket_id):
...
+            label = 'attachment: ' + filename
+
         if keep_trac_ticket_references:
             url = '%s/ticket/%s/%s' % (trac_url_attachment, str(self._ticket_id), filename)
         else:
             # FIXME: This uses the global variable dest
             a, _, _ = gh_create_attachment(dest, None, filename, self._ticket_id, None)
-            url = a.url
-            if url.endswith('.gz'):
+            if a.url.endswith('.gz'):
```
Comment (mkoeppe) [inline migrate.py:1370]: Is this change correct? `url` is used at the end of the function

Comment (mkoeppe) [reply migrate.py:1370]: Oh now I see.

**Comment by kwankyu** [comment]
While at it, I want to remove the comment `# FIXME: This uses the global variable dest` scattered throughout. If I understand correctly, in our mode of operation, the `dest` variable is always `None` as `must_convert_issues` is `False`. Am I right?

**Comment by mkoeppe** [comment]
No, `dest` is also set here: https://github.com/sagemath/trac-to-github/blob/master/migrate.py#L2684

**Comment by mkoeppe** [comment]
And `must_convert_issues` is True

**Comment by kwankyu** [comment]
The `gh_create_attachement` just below the `FIXME` comment really create an attachment at `dest` or is it dummy?

**Comment by mkoeppe** [comment]
In our mode of operation (writing a migration archive), `dest` is only in memory. There is no connection to GitHub.com. 
The `MigrationArchiveWritingRequester` writes stuff that would be sent to GitHub.com to files.

**Comment by kwankyu** [comment]
OK. Then it seems there is nothing to "fix". May I just get rid of `FIXME` comment?

**Comment by mkoeppe** [comment]
The changes here on the PR break some links:
`````
diff --git a/Issues-33xxx/33299.md b/Issues-33xxx/33299.md
index 0e26367..2266483 100644
--- a/Issues-33xxx/33299.md
+++ b/Issues-33xxx/33299.md
@@ -50,11 +50,11 @@ B = b.plot3d(z=2)
 A + B
 ```
 
-![](https://doc.sagemath.org/html/en/reference/plotting/_images/bezier_path-2.svg)
+![](https://github.com/sagemath/sage/files/ticket33299/ddbe2f82adc283435e7bcba665271842.svg)
 
 ## good image0
 
-![](ticket:33299:a0.2.png)
+![](https://github.com/sagemath/sage/files/ticket33299/cc98f85cef8ad8d0304fd78b260040a8.png)
 
 ## EXAMPLE
 
`````

**Comment by mkoeppe** [comment]
> Then it seems there is nothing to "fix". May I just get rid of `FIXME` comment?

The FIXME is just regarding the stylistic break of referring to this value via a global variable rather than via argument passing. The code is correct as is. OK to remove the comment.

**Comment by mkoeppe** [comment]
That fixes it, thanks!

Here's another bad link:
`````
diff --git a/Issues-33xxx/33271.md b/Issues-33xxx/33271.md
index 178b2bb..a71af5b 100644
--- a/Issues-33xxx/33271.md
+++ b/Issues-33xxx/33271.md
@@ -211,8 +211,8 @@ Description changed:
  ```
  
  ## output image1
--![](ticket::tmp_m0fft6r1.png)
-+![](ticket:33271:tmp_m0fft6r1.png)
+-![](https://github.com/sagemath/sage/files/ticket/tmp_m0fft6r1.png)
++![](https://github.com/sagemath/sage/files/ticket33271/tmp_m0fft6r1.png)
  
  ## expected image1
  ![](https://doc.sagemath.org/html/en/reference/plotting/_images/density_plot-6.svg)
``````

**Comment by kwankyu** [comment]
If the name of an attachment ends with `.gz`, then url is correctly constructed, but the label still uses the original name without `.gz`. This is because of my code change. Is this okay or needs to be reverted?

**Comment by kwankyu** [comment]
> That fixes it, thanks!
> 
> Here's another bad link:
> 
> ```
> diff --git a/Issues-33xxx/33271.md b/Issues-33xxx/33271.md
> index 178b2bb..a71af5b 100644
> --- a/Issues-33xxx/33271.md
> +++ b/Issues-33xxx/33271.md
> @@ -211,8 +211,8 @@ Description changed:
>   ```
>   
>   ## output image1
> --
> -+
> +-
> ++
>   
>   ## expected image1
>   
> ```

Which one? 

`ticket::tmp_m0fft6r1.png` seems a wrong form written by the author.

**Comment by mkoeppe** [comment]
> `ticket::tmp_m0fft6r1.png` seems a wrong form written by the author.

That's the one I meant. Sorry, I forgot to check whether the original works correctly.

**Comment by mkoeppe** [comment]
> If the name of an attachment ends with `.gz`, then url is correctly constructed, but the label still uses the original name without `.gz`. This is because of my code change. Is this okay

I'm OK with this change

**Comment by mkoeppe** [comment]
Thank you!

**Comment by kwankyu** [comment]
Thanks!

**Pull request merged**